domain/application/service/unit.go

```diff
@@ -564,23 +573,36 @@ func (s *Service) GetUnitWorkloadVersion(ctx context.Context, unitName coreunit.
 // - [uniterrors.UnitNotFound] if the unit does not exist
 // - [network.NoAddressError] if the unit has no public address associated
 func (s *Service) GetUnitPublicAddress(ctx context.Context, unitName coreunit.Name) (network.SpaceAddress, error) {
-	unitUUID, err := s.st.GetUnitUUIDByName(ctx, unitName)
+	publicAddresses, err := s.GetUnitPublicAddresses(ctx, unitName)
 	if err != nil {
 		return network.SpaceAddress{}, errors.Capture(err)
 	}
+	return publicAddresses[0], nil
```

Comment:
Worth adding a length > 0 check?

Reply:
No need, since `GetUnitPublicAddresses()` returns a `NoAddressError` if the length is 0.